System/Remote/Monitoring/Statsd.hs

```diff
@@ -146,22 +158,32 @@ forkStatsd opts store = do
 loop :: Metrics.Store            -- ^ Metric store
      -> Metrics.Sample           -- ^ Last sampled metrics
      -> (B8.ByteString -> IO ()) -- ^ Action to send a sample
+     -> Int                      -- ^ The current iteration
      -> StatsdOptions            -- ^ Options
      -> IO ()
-loop store lastSample sendSample opts = do
+loop store lastSample sendSample currentIteration opts = do
     start <- time
-    sample <- Metrics.sampleAll store
-    let !diff = diffSamples lastSample sample
-    flushSample diff sendSample opts
+    currentSample <- Metrics.sampleAll store
+    -- Check wether or not we need to perform a "full flush", in
+    -- which case we reset the counter and send downstream to flushing
+    -- the current snapshot of the `Store`, otherwise we increase the counter
+    -- and compute the diff between the previous and the current sample.
+    let (!newIterationCount, !sample) =
+          case fullFlushIterations opts of
+            Just x | x >= currentIteration -> (0, sample)
+            _  -> (newIterationCount + 1, diffSamples lastSample currentSample)
```

Comment:
> `newIterationCount + 1`

Hmm, shouldn't this be `currentIteration + 1`?

Reply:
Argh, it totally should be, sorry.